### PR TITLE
BE-84 Tree shake the config

### PR DESCRIPTION
**Pull request by chiibis**
### References
[help]: # (Add link/s to every reference related to this pull request: issue, migrations, another PRs, ...)
* **Issue:** https://hellogetsafe.atlassian.net/browse/BE-84
* **Related pull-requests:**
* **Sentry errors:**

### What is the goal? How is it being implemented?
[help]: # (Provide a description of the overall goal and a description of the implementation.)
Since we enabled all rules by default in 2.0.0, now we have bunch of redundant configuration. Let's clean it up

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# v2.3.0 2024-02-09
```
Comment (alecspopa) [inline CHANGELOG.md:1]: What happened to 2.2.0?

File: default.yml [hunk 2]
```diff
@@ -29,26 +29,19 @@ AllCops:
   DisabledByDefault: false
   NewCops: disable # to avoid warnings. TODO: remove this setting and add new cops from the warning.
 
-# A bunch of these layout cops are enabled by default,
-# but for some reason 'DisableByDefault' is set to 'true' above.
```
Comment (alecspopa) [inline default.yml:33]: This comment is not valid anymore (based on line 29)

File: default.yml [hunk 3]
```diff
@@ -511,156 +406,4 @@ Naming/FileName:
...
-#  Enabled: false
-#Style/EmptyMethod:
-#  Enabled: false
-#Layout/EmptyLineBetweenDefs:
-#  Enabled: false
-#Naming/ConstantName:
-#  Enabled: false
-#Rails/RelativeDateConstant:
-#  Enabled: false
+  Enabled: false
```
Comment (alecspopa) [inline default.yml:409]:
```suggestion
  Enabled: false

```

**Review by alecspopa — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (chiibis) [reply CHANGELOG.md:1]: There's an open PR with 2.2.0 https://github.com/nxt-insurance/nxt_cop/pull/132/files

I thought it'll be merged before mine. But I'm not sure now

File: default.yml (hunk 2 above)
Comment (chiibis) [reply default.yml:33]: I've removed these lines 

Do you want me to remove the rest of the comment too?

Comment (alecspopa) [reply default.yml:33]: If the comment does not make sense anymore and you're making changes around it, sure :)

File: CHANGELOG.md (hunk 1 above)
Comment (TylerWitt) [reply CHANGELOG.md:1]: I'll merge it!

Comment (chiibis) [reply CHANGELOG.md:1]: Rebased

**Pull request merged**